rivers.py

```
@@ -0,0 +1,302 @@
+    start_point = QgsPointXY(polyline[0])
+    end_point = QgsPointXY(polyline[-1])
+
+    # Высотные данные начальной точки
+    start_z = dem_layer.dataProvider().identify(start_point, QgsRaster.IdentifyFormatValue)
+    start_z_idx = line_provider.fields().indexOf('start_z')
+    if start_z.isValid():
+        start_z_value = start_z.results()[1]
+        feature['start_z'] = start_z_value
+
+    # Высотные данные конечной точки
+    end_z = dem_layer.dataProvider().identify(end_point, QgsRaster.IdentifyFormatValue)
```

Ruff (0.7.0)

Undefined name `QgsRaster`

(F821)